mobile/src/components/Navigator.tsx

```diff
@@ -17,6 +18,7 @@ const Navigator = () => {
       {isLoggedIn ? (
         <>
           <Stack.Screen name="HomeScreen" component={HomeScreen} />
+          <Stack.Screen name="LoadingScreen" component={LoadingScreen} />
```

Comment:
We can try to handle the back button behavior by using [this](https://reactnavigation.org/docs/preventing-going-back/) reference. I think if we just prevent the default behavior it will prevent all back buttons and gestures.